cmd/state-installer/installer.go

```diff
@@ -226,3 +229,30 @@ func isStateExecutable(name string) bool {
 	}
 	return false
 }
+
+func installedOnPath(installRoot string) (bool, string, error) {
+	if !fileutils.DirExists(installRoot) {
+		return false, "", nil
+	}
+
+	path := appinfo.StateApp(installRoot).Exec()
+	if fileutils.TargetExists(path) {
+		return true, filepath.Dir(path), nil
+	}
+
+	found, err := fileutils.FindFileInPath(installRoot, installation.InstallDirMarker)
```

Comment:
I don't think this is where we should be using the InstallDirMarker. I would see it used in places like `appinfo.newAppInfo()` and `installation.InstallPath()`. This function is just checking whether the resulting install path has state tool on it.

Reply:
That's fair. I added it because of this comment: https://github.com/ActiveState/cli/pull/1784#discussion_r842183809 but the logic was different at the time.